### PR TITLE
[Augur] Expose Augur versions 

### LINKED ISSUE #685: [Augur] Add version tracking for tree software + iqtree model
:cool:

### :pushpin: Explain the Request

Add capturing versions for iqtruee, fasttree, raxml, and verson of iqtree model being used. Per @kapsakcj

### PR DESCRIPTION
**Pull request by Michal-Babins**
This PR closes #685 

🗑️ This dev branch should <NOT> be deleted after merging to main.

## :brain: Summary

This PR captures versions for iqtree, fasttree, raxml, and version of iqtree model being used with Augur. 

## :zap: Impacted Workflows/Tasks

`wf_augur`
`task_augur_tree`

This PR may lead to different results in pre-existing outputs: **No**

This PR uses an element that could cause duplicate runs to have different results: **No**

## :hammer_and_wrench: Changes

Outputs have been added to capture the following version outputs for `augur`:

```
String iqtree_version = read_string("IQTREE_VERSION")
String fasttree_version = read_string("FASTTREE_VERSION")
String raxml_version = read_string("RAXML_VERSION")
String iqtree_model_used = read_string("FINAL_MODEL.txt")
```

Where `iqtree_model_used` is for the actual iqtree model and not iqtree software version which is captured in `iqtree_version`.

### :gear: Algorithm

In `task_augur_tree`, now depending on what `method` is chosen (iqtree,fasttree,raxml) that sofware version will be captured in the output. 

### ⬅️ Outputs

The following output has been added to `task_augur_tree`:
```
String iqtree_version = read_string("IQTREE_VERSION")
String fasttree_version = read_string("FASTTREE_VERSION")
String raxml_version = read_string("RAXML_VERSION")
String iqtree_model_used = read_string("FINAL_MODEL.txt")
```
The following outputs have been added top level to `wf_augur`:
```
String augur_iqtree_model_used = augur_tree.iqtree_model_used
String augur_iqtree_version = augur_tree.iqtree_version
String augur_fasttree_version = augur_tree.fasttree_version
String augur_raxml_version = augur_tree.raxml_version
```
## :microscope: Final Developer Checklist

- [x] The workflow/task has been tested and results, including file contents, are as anticipated
- [x] The CI/CD has been adjusted and tests are passing (Theiagen developers)
- [x] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] Documentation and/or workflow diagrams have been updated if applicable (Theiagen developers only)

## 🎯 Reviewer Checklist

- [ ] All changed results have been confirmed
- [x] You have tested the PR appropriately (see the [testing guide](https://theiagen.notion.site/PR-Testing-Guide-Determining-Appropriate-Levels-of-Testing-4764e98a6aeb460185039c0896714590) for more information)
- [x] All code adheres to the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] MD5 sums have been updated
- [x] The PR author has addressed all comments
- [x] The documentation has been updated

### CODE REVIEW

**Comment by kapsakcj** [comment]
For clarity - `iqtree_model_used` output String was implemented/added in a separate PR: https://github.com/theiagen/public_health_bioinformatics/pull/674

This PR adds the ability to capture the model used in the rare scenario that the user provides a BED file to `augur_tree` task for masking. This was not working previously.

Want to also note that this PR adds `augur_mafft_version` to the `wf_augur` top level workflow.

And lastly, we added `set -euo pipefail` to the `augur_tree` and `augur_align` tasks.

Here are links to my tests in Terra:

- ✅ testing new String outputs `augur_raxml_version`  & `augur_mafft_version` output. 
  - `augur_fasttree_version`, `augur_iqtree_version`, and `augur_iqtree_model_used` are blank as expected. 
  - https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Support_Sandbox/job_history/f1c8ce59-d7dc-4a01-9b81-0be7f9a98bf4
- ✅ testing new String outputs `augur_fasttree_version` & `augur_mafft_version`. 
  - `augur_raxml_version`, `augur_iqtree_version`, and `augur_iqtree_model_used` are blank as expected. 
  - https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Support_Sandbox/job_history/802227b1-cefb-44df-9670-500111ecca71
- ✅ testing new String output `augur_iqtree_version`. Ensuring (somewhat new) String output `augur_iqtree_model_used` is output
  - `augur_fasttree_version` and `augur_raxml_version` are blank as expected.
  - https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Support_Sandbox/job_history/5f1f3aa0-e904-4e03-a6b9-a58eea460de4

I'm going to launch one more test with a BED file used as input for the augur_tree task, just to ensure iqtree version and model output strings are still present

**Comment by kapsakcj** [comment]
last test. Ran successfully ✅ and produced the `augur_iqtree_model_used` even when using a BED file as input to the `exclude_sites` variable for the `augur_tree` task for masking. https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Support_Sandbox/job_history/5e89f9f8-221e-454b-965a-5be714810a10

Don't forget about updating the Augur docs for these new output string

**Review by kapsakcj — APPROVED** [review]
Thanks for running those tests and updating the docs, much appreciated! Nice work on this PR 👍 

CC @sage-wright I believe it's ready for merging

File: tasks/phylogenetic_inference/augur/task_augur_align.wdl [hunk 1]
```diff
@@ -12,8 +12,13 @@ task augur_align {
     String docker = "us-docker.pkg.dev/general-theiagen/biocontainers/augur:22.0.2--pyhdfd78af_0"
   }
   command <<<
+    set -euo pipefail
+    
     # capture version information
     augur version > VERSION
+    echo
```
Comment (sage-wright) [inline tasks/phylogenetic_inference/augur/task_augur_align.wdl:19]: you've got an extra echo here

Comment (sage-wright) [reply tasks/phylogenetic_inference/augur/task_augur_align.wdl:19]: ah i see you're using them as line breaks in the output log

**Review by sage-wright — APPROVED** [review]
⭐

**Pull request merged**